File: FinAnalyzer.Domain/Entities/Category.cs
namespace FinAnalyzer.Domain.Entities;

public class Category : BaseEntity
{
    public string Name { get; set; }

    public string? Description { get; set; }

    /// <summary>
    /// true, если категория расходная
    /// </summary>
    public bool IsExpenditure { get; set; }

    public string IconAuthor { get; set; }

    public string IconName { get; set; }

    public string Color { get; set; }

    public Room Room { get; set; }
    public int RoomId { get; set; }
}

File: mobile/app/api/interfaces/category.ts
import { IconAuthor } from '../../components/ui/Icon';

export interface ICategory {
  id: number;
  name: string;
  description: string | null;
  isExpenditure: boolean;
  iconAuthor: IconAuthor;
  iconName: string;
  iconColor: string;
}

export interface ICategoryCreate {
  name: string;
  description?: string;
  isExpenditure: boolean;
  iconAuthor?: IconAuthor;
  iconName: string;
  iconColor: string;
}

export interface ICategoryUpdate {
  id: number;
  name: string;
  description?: string;
  isExpenditure: boolean;
  iconAuthor?: IconAuthor;
  iconName: string;
  iconColor: string;
}
